src/commonMain/kotlin/fr/acinq/lightning/io/Peer.kt

```diff
@@ -427,6 +419,39 @@ class Peer(
         input.send(cmd)
     }
 
+    /**
+     * Do a splice out using any suitable channel
+     * @return  [Command.Splice.Companion.Result] if a splice was attempted, or {null} if no suitable
+     *          channel was found
+     */
+    suspend fun spliceOut(amount: Satoshi, scriptPubKey: ByteVector, feeratePerKw: FeeratePerKw): Command.Splice.Response? {
+        return channels.values
+            .filterIsInstance<Normal>()
+            .firstOrNull { it.commitments.availableBalanceForSend() > amount }
+            ?.let { channel ->
+                val spliceCommand = Command.Splice.Request(
+                    replyTo = CompletableDeferred(),
```

Comment:
I think it's best to leave it that way (discussed with @dpad85 too). The beauty of kotlin's `suspend` functions is that their signature appears synchronous. We would break this by exposing the `replyTo`. It is trivial for integrators to reverse the pattern if they need to.